apis/util/utils.go

```diff
@@ -14,18 +14,27 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package validation
+package utils
```

Comment:
All versioned code is under `apis/<version>`. This file seems to be specific to v1a2 but we are moving it out.
@robscott Any guidance on the upstream practice here?

Reply:
Hey, Harry, here is the root source https://github.com/kubernetes-sigs/gateway-api/pull/780#discussion_r691482077

Reply:
are we good with this one ? can we close this ?

Reply:
Good question, this is a large change. I think we will inevitably need a set of util funcs (already suggested by https://github.com/kubernetes-sigs/gateway-api/issues/826). My preference would be to put all utils into `pkg/utils` instead of `apis/utils` though.

Reply:
9be59b9deca99e1af9fbdab5fc6a20a90bacd7dc